node/silkworm/db/storage.hpp

```diff
@@ -0,0 +1,184 @@
+/*
+    Copyright 2021 The Silkworm Authors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+#ifndef SILKWORM_STORAGE_HPP_
+#define SILKWORM_STORAGE_HPP_
+
+#include "mdbx.hpp"
+#include "tables.hpp"
+
+namespace silkworm::db {
+
+//! \brief Holds the storage mode set
+struct StorageMode {
+    bool Initialized;  // Whether db storage has been initialized
```

Comment:
1) Please use snake_case (e.g. `call_traces`) for struct data members for consistency with [Goggle's Style Guide](https://google.github.io/styleguide/cppguide.html#Variable_Names) and the rest of our code.
2) I strongly encourage to use default initializers for things that otherwise can be uninitialized. For example, if we have
```
struct Foo {
 bool a;
 bool b;
};
```
and later somebody writes `Foo x;` then happy debugging of inconsistent behaviour on different platforms 😖
```
struct Foo {
 bool a{false};
 bool b{false};
};
```
instead prevents such mistakes.
3. Given the above, are `Initialized` & `kDefaultStorageMode` necessary? What if we have something like
```
struct StorageMode {
    bool history{true};
    bool receipts{true};
    bool tx_index{true};
    bool call_traces{true};
    bool tevm{false};
}
```
4. The same goes for `PruneMode`.

Reply:
I would not focus too much on StorageMode as it has to be dropped (Erigon has switched to PruneMode)
Will look at PruneMode